### ISSUE TITLE
Add Regex lookup expression FilterSet fields to autogenerated list

### ISSUE THREAD

**Issue by bryanculver** [nautobot/nautobot#1525]
Fields that automagically generate character-based lookup expressions should also generate regex-supporting lookup expressions.

**Comment by jathanism** [comment]
This one is super easy!

```diff
diff --git a/nautobot/utilities/constants.py b/nautobot/utilities/constants.py
index 44d018038..2e547b65e 100644
--- a/nautobot/utilities/constants.py
+++ b/nautobot/utilities/constants.py
@@ -12,6 +12,8 @@ FILTER_CHAR_BASED_LOOKUP_MAP = dict(
     nisw="istartswith",
     ie="iexact",
     nie="iexact",
+    r="regex",
+    nr="regex",
 )

 FILTER_NUMERIC_BASED_LOOKUP_MAP = dict(n="exact", lte="lte", lt="lt", gte="gte", gt="gt")
(END)
```

**Comment by jathanism** [comment]
Checklist:

- [ ] Update lookup map constants
- [ ] Update unit tests
- [ ] Revise documentation

**Comment by jathanism** [comment]
Checklist:

- [x] Update lookup map constants
- [x] Update unit tests
- [ ] Revise documentation

**Comment by bryanculver** [comment]
Closed with #1535
